libs/dyn/mapping.go

```diff
@@ -91,28 +90,22 @@ func (m *Mapping) GetByString(skey string) (Value, bool) {
 }
 
 // Set sets the value for the given key in the mapping.
-// If the key already exists, the value is updated.
+// If the key already exists, the value is updated. The location loc is ignored.
 // If the key does not exist, a new key-value pair is added.
-// The key must be a string, otherwise an error is returned.
-func (m *Mapping) Set(key, value Value) error {
-	skey, ok := key.AsString()
-	if !ok {
-		return fmt.Errorf("key must be a string, got %s", key.Kind())
-	}
-
+func (m *Mapping) SetLoc(skey string, loc []Location, value Value) {
 	// If the key already exists, update the value.
 	if i, ok := m.index[skey]; ok {
-		m.pairs[i].Value = value
-		return nil
+		p := &m.pairs[i]
+		p.Value = value
```

Comment:
Looks equivalent to original assignment.

Reply:
thanks, restored in 25210ddabf0a7284a21f3d2f966f53b6f9f4fed5